scene/2d/light_2d.cpp

```diff
@@ -449,7 +476,7 @@ void PointLight2D::_bind_methods() {
 	ClassDB::bind_method(D_METHOD("set_texture_scale", "texture_scale"), &PointLight2D::set_texture_scale);
 	ClassDB::bind_method(D_METHOD("get_texture_scale"), &PointLight2D::get_texture_scale);
 
-	ADD_PROPERTY(PropertyInfo(Variant::OBJECT, "texture", PROPERTY_HINT_RESOURCE_TYPE, "Texture2D"), "set_texture", "get_texture");
+	ADD_PROPERTY(PropertyInfo(Variant::OBJECT, "texture", PROPERTY_HINT_RESOURCE_TYPE, "CompressedTexture2D,PortableCompressedTexture2D,ImageTexture,CurveTexture,CurveXYZTexture,GradientTexture1D,GradientTexture2D,PlaceholderTexture2D,NoiseTexture2D"), "set_texture", "get_texture");
```

Comment:
This list of types is something we should probably reuse across in the editor a lot more, as there are a *lot* of situations where a type like CameraTexture or AtlasTexture is not allowed (only one or two of those at at time).

Maybe we should have a special syntax using `-` to denote exceptions, like `Texture2D,-ViewportTexture,-AtlasTexture` to allow anything that extends Texture2D *except* ViewportTexture and AtlasTexture.

Reply:
I did not even know this was a feature. Concerning to me. It has been here for a while, yet PROPERTY_HINT_RESOURCE_TYPE is not documented well enough to showcase this.

Reply:
> Maybe we should have a special syntax using - to denote exceptions, like Texture2D,-ViewportTexture,-AtlasTexture to allow anything that extends Texture2D except ViewportTexture and AtlasTexture.

That sounds more scalable in case new texture types are added. I think the hint is only used in EditorResourcePicker, so it should be trivial to implement (the picker already unfolds the base type into inheriting classes).

Reply:
Honestly I'm not sure I like this syntax to exclude unsupported types. We're excluding as many times as we'd have to explicitly include using the usual syntax.

And I think future proofing the addition of new types is not a good idea, as we don't know whether those new types would be supported. So it's better to state explicitly what we know we support. This would also make it easier possibly in the future to actually parse this hint and provide this information automatically in the class reference for example.